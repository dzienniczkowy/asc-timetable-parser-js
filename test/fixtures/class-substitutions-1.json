{
  "r": "<div class=\"_RSS365\"><style type=\"text/css\">._RSS365 .signature {font-Size:20px;float:right;border-Bottom:1px dashed #c0c0c0;width:100px;margin-Right:4px;margin-Left:10px}\n._RSS365 .row {display:flex;flex-Direction:row;align-Items:baseline;font-Size:12px;margin-Bottom:1px}\n._RSS365 .period {font-Size:16px;text-Align:center;flex-Grow:0;margin-Right:10px}\n._RSS365 .period>span {white-Space:pre;display:inline-block;min-Width:30px}\n._RSS365 .info {flex:1}\n._RSS365 .section {margin-Bottom:15px;border:1px solid #c0c0c0;border-Radius:6px;padding:4px}\n._RSS365 .header {font-Size:20px;margin-Bottom:5px;font-Weight:bold;border:2px solid #c0c0c0;border-Radius:6px;padding:4px;display:inline-block}\n._RSS365 .rows {padding-Left:5px}\n._RSS365 .rows .signature {font-Size:inherit}</style><div data-date=\"2022-09-12\"><div style=\"text-align:center\"><span class=\"print-font-resizable\">Absent teachers: Szczygieł Robert , Kalinowski Gracja , Jakubiak Florencja , Wypych Walerian , Rybarczyk Kalistrat , Biernacki Bazyli </span></div><div style=\"text-align:center\"><span class=\"print-font-resizable\">Absent classes: 1A , 1E , 2C (0 - 4) </span></div><div style=\"height:15px;flex:none\"></div><div class=\"section print-nobreak\"><div class=\"header\"><span class=\"print-font-resizable\">1A</span></div><div class=\"rows\"><div class=\"row absent\"><div class=\"period\"><span class=\"print-font-resizable\">whole day</span></div><div class=\"info\"><span class=\"print-font-resizable\"><img src=\"/global/pics/ui/absent_32.svg\" style=\"height:16px;display:inline-block;vertical-align:text-bottom;margin-right:5px\"/>Absent</span></div></div><div class=\"row remove\"><div class=\"period\"><span class=\"print-font-resizable\">(3)</span></div><div class=\"info\"><span class=\"print-font-resizable\">historia i teraźniejszość - Jędrzejczyk Krystian, Cancelled</span></div></div><div class=\"row remove\"><div class=\"period\"><span class=\"print-font-resizable\">(4)</span></div><div class=\"info\"><span class=\"print-font-resizable\">matematyka - Bielski Hilary, Cancelled</span></div></div><div class=\"row remove\"><div class=\"period\"><span class=\"print-font-resizable\">(5 - 6)</span></div><div class=\"info\"><span class=\"print-font-resizable\">język polski - Pietruszka Dawid, Cancelled</span></div></div></div></div><div class=\"section print-nobreak\"><div class=\"header\"><span class=\"print-font-resizable\">1D</span></div><div class=\"rows\"><div class=\"row change\"><div class=\"period\"><span class=\"print-font-resizable\">4</span></div><div class=\"info\"><span class=\"print-font-resizable\">biologia - Substitution: (Kalinowski Gracja) ➔ Guzik Noemi</span></div></div></div></div><div class=\"section print-nobreak\"><div class=\"header\"><span class=\"print-font-resizable\">1E</span></div><div class=\"rows\"><div class=\"row absent\"><div class=\"period\"><span class=\"print-font-resizable\">whole day</span></div><div class=\"info\"><span class=\"print-font-resizable\"><img src=\"/global/pics/ui/absent_32.svg\" style=\"height:16px;display:inline-block;vertical-align:text-bottom;margin-right:5px\"/>Absent</span></div></div><div class=\"row remove\"><div class=\"period\"><span class=\"print-font-resizable\">(1 - 2)</span></div><div class=\"info\"><span class=\"print-font-resizable\">uniwersytecki fakultet matematuczny - Gawron Atanazy, Cancelled</span></div></div><div class=\"row remove\"><div class=\"period\"><span class=\"print-font-resizable\">(3 - 4)</span></div><div class=\"info\"><span class=\"print-font-resizable\">algorytmika_1: algorytmika - Łuczak Dina, Cancelled</span></div></div><div class=\"row remove\"><div class=\"period\"><span class=\"print-font-resizable\">(3 - 4)</span></div><div class=\"info\"><span class=\"print-font-resizable\">informatyka_2: informatyka - Jakubiak Florencja, Cancelled</span></div></div><div class=\"row remove\"><div class=\"period\"><span class=\"print-font-resizable\">(5 - 6)</span></div><div class=\"info\"><span class=\"print-font-resizable\">informatyka_1: informatyka - Jakubiak Florencja, Cancelled</span></div></div><div class=\"row remove\"><div class=\"period\"><span class=\"print-font-resizable\">(5 - 6)</span></div><div class=\"info\"><span class=\"print-font-resizable\">algorytmika_2: algorytmika - Łuczak Dina, Cancelled</span></div></div></div></div><div class=\"section print-nobreak\"><div class=\"header\"><span class=\"print-font-resizable\">1F</span></div><div class=\"rows\"><div class=\"row remove\"><div class=\"period\"><span class=\"print-font-resizable\">(3)</span></div><div class=\"info\"><span class=\"print-font-resizable\">język polski - Biernacki Bazyli, Cancelled</span></div></div><div class=\"row change\"><div class=\"period\"><span class=\"print-font-resizable\">4</span></div><div class=\"info\"><span class=\"print-font-resizable\">(język polski) ➔ historia - Substitution: (Biernacki Bazyli) ➔ Polak Apollinary, za drugą lekcję</span></div></div></div></div><div class=\"section print-nobreak\"><div class=\"header\"><span class=\"print-font-resizable\">2A</span></div><div class=\"rows\"><div class=\"row change\"><div class=\"period\"><span class=\"print-font-resizable\">2</span></div><div class=\"info\"><span class=\"print-font-resizable\">historia - Substitution: (Szczygieł Robert) ➔ Turek Laura, zł. grup</span></div></div></div></div><div class=\"section print-nobreak\"><div class=\"header\"><span class=\"print-font-resizable\">2D</span></div><div class=\"rows\"><div class=\"row change\"><div class=\"period\"><span class=\"print-font-resizable\">7</span></div><div class=\"info\"><span class=\"print-font-resizable\">biologia - Substitution: (Kalinowski Gracja) ➔ Polak Apollinary, zł. grup</span></div></div></div></div><div class=\"section print-nobreak\"><div class=\"header\"><span class=\"print-font-resizable\">2E</span></div><div class=\"rows\"><div class=\"row remove\"><div class=\"period\"><span class=\"print-font-resizable\">(8)</span></div><div class=\"info\"><span class=\"print-font-resizable\">grupa1: informatyka - Jakubiak Florencja, Cancelled</span></div></div></div></div><div class=\"section print-nobreak\"><div class=\"header\"><span class=\"print-font-resizable\">2F</span></div><div class=\"rows\"><div class=\"row change\"><div class=\"period\"><span class=\"print-font-resizable\">2</span></div><div class=\"info\"><span class=\"print-font-resizable\">język polski - Substitution: (Biernacki Bazyli) ➔ Jurkiewicz Dionizy</span></div></div></div></div><div class=\"section print-nobreak\"><div class=\"header\"><span class=\"print-font-resizable\">2G</span></div><div class=\"rows\"><div class=\"row change\"><div class=\"period\"><span class=\"print-font-resizable\">3</span></div><div class=\"info\"><span class=\"print-font-resizable\">matematyka - Substitution: (Wypych Walerian) ➔ Słowiński Szczepan</span></div></div><div class=\"row remove\"><div class=\"period\"><span class=\"print-font-resizable\">(8)</span></div><div class=\"info\"><span class=\"print-font-resizable\">biologia - Kalinowski Gracja, Cancelled</span></div></div></div></div><div class=\"section print-nobreak\"><div class=\"header\"><span class=\"print-font-resizable\">3A</span></div><div class=\"rows\"><div class=\"row remove\"><div class=\"period\"><span class=\"print-font-resizable\">(1 - 2)</span></div><div class=\"info\"><span class=\"print-font-resizable\">religia - Rybarczyk Kalistrat, Cancelled</span></div></div><div class=\"row change\"><div class=\"period\"><span class=\"print-font-resizable\">6</span></div><div class=\"info\"><span class=\"print-font-resizable\">(biologia) ➔ fizyka - Substitution: (Kalinowski Gracja) ➔ Kulik Olaf</span></div></div></div></div><div class=\"section print-nobreak\"><div class=\"header\"><span class=\"print-font-resizable\">3B</span></div><div class=\"rows\"><div class=\"row remove\"><div class=\"period\"><span class=\"print-font-resizable\">(1 - 2)</span></div><div class=\"info\"><span class=\"print-font-resizable\">religia - Rybarczyk Kalistrat, Cancelled</span></div></div></div></div><div class=\"section print-nobreak\"><div class=\"header\"><span class=\"print-font-resizable\">3C</span></div><div class=\"rows\"><div class=\"row change\"><div class=\"period\"><span class=\"print-font-resizable\">3</span></div><div class=\"info\"><span class=\"print-font-resizable\">religia - Substitution: (Rybarczyk Kalistrat) ➔ Jurek Arseniusz</span></div></div><div class=\"row change\"><div class=\"period\"><span class=\"print-font-resizable\">5</span></div><div class=\"info\"><span class=\"print-font-resizable\">historia - Substitution: (Szczygieł Robert) ➔ Pietruszka Dawid, za nieobecny oddział</span></div></div></div></div><div class=\"section print-nobreak\"><div class=\"header\"><span class=\"print-font-resizable\">3E</span></div><div class=\"rows\"><div class=\"row remove\"><div class=\"period\"><span class=\"print-font-resizable\">(1)</span></div><div class=\"info\"><span class=\"print-font-resizable\">matematyka - Wypych Walerian, Cancelled, historia odwołana</span></div></div><div class=\"row remove\"><div class=\"period\"><span class=\"print-font-resizable\">(2)</span></div><div class=\"info\"><span class=\"print-font-resizable\">matematyka - Wypych Walerian, Cancelled</span></div></div></div></div><div class=\"section print-nobreak\"><div class=\"header\"><span class=\"print-font-resizable\">3F</span></div><div class=\"rows\"><div class=\"row change\"><div class=\"period\"><span class=\"print-font-resizable\">6</span></div><div class=\"info\"><span class=\"print-font-resizable\">historia - Substitution: (Szczygieł Robert) ➔ Wójcik Aleksandra</span></div></div><div class=\"row change\"><div class=\"period\"><span class=\"print-font-resizable\">7</span></div><div class=\"info\"><span class=\"print-font-resizable\">historia - Substitution: (Szczygieł Robert) ➔ Wysocki Harald, zł. grup</span></div></div></div></div><div class=\"section print-nobreak\"><div class=\"header\"><span class=\"print-font-resizable\">3G</span></div><div class=\"rows\"><div class=\"row change\"><div class=\"period\"><span class=\"print-font-resizable\">7</span></div><div class=\"info\"><span class=\"print-font-resizable\">religia - Substitution: (Rybarczyk Kalistrat) ➔ Pietruszka Dawid, zł. grup</span></div></div></div></div><div class=\"section print-nobreak\"><div class=\"header\"><span class=\"print-font-resizable\">4H</span></div><div class=\"rows\"><div class=\"row remove\"><div class=\"period\"><span class=\"print-font-resizable\">(1)</span></div><div class=\"info\"><span class=\"print-font-resizable\">historia - Szczygieł Robert, Cancelled, grupa z bilogii o 7.10- czytelnia</span></div></div></div></div></div></div>"
}
