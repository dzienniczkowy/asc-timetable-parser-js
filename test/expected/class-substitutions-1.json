{
  "date": "2022-09-12",
  "absentClasses": [
    {
      "name": "1A",
      "lessons": null
    },
    {
      "name": "1E",
      "lessons": null
    },
    {
      "name": "2C",
      "lessons": {
        "first": 0,
        "last": 4
      }
    }
  ],
  "absentTeachers": [
    {
      "name": "Szczygieł Robert",
      "lessons": null
    },
    {
      "name": "Kalinowski Gracja",
      "lessons": null
    },
    {
      "name": "Jakubiak Florencja",
      "lessons": null
    },
    {
      "name": "Wypych Walerian",
      "lessons": null
    },
    {
      "name": "Rybarczyk Kalistrat",
      "lessons": null
    },
    {
      "name": "Biernacki Bazyli",
      "lessons": null
    }
  ],
  "sections": [
    {
      "name": "1A",
      "changes": []
    },
    {
      "name": "1D",
      "changes": [
        {
          "lessons": {
            "first": 4,
            "last": 4
          },
          "info": "biologia - Substitution: (Kalinowski Gracja) ➔ Guzik Noemi"
        }
      ]
    },
    {
      "name": "1E",
      "changes": []
    },
    {
      "name": "1F",
      "changes": [
        {
          "lessons": {
            "first": 4,
            "last": 4
          },
          "info": "(język polski) ➔ historia - Substitution: (Biernacki Bazyli) ➔ Polak Apollinary, za drugą lekcję"
        }
      ]
    },
    {
      "name": "2A",
      "changes": [
        {
          "lessons": {
            "first": 2,
            "last": 2
          },
          "info": "historia - Substitution: (Szczygieł Robert) ➔ Turek Laura, zł. grup"
        }
      ]
    },
    {
      "name": "2D",
      "changes": [
        {
          "lessons": {
            "first": 7,
            "last": 7
          },
          "info": "biologia - Substitution: (Kalinowski Gracja) ➔ Polak Apollinary, zł. grup"
        }
      ]
    },
    {
      "name": "2E",
      "changes": []
    },
    {
      "name": "2F",
      "changes": [
        {
          "lessons": {
            "first": 2,
            "last": 2
          },
          "info": "język polski - Substitution: (Biernacki Bazyli) ➔ Jurkiewicz Dionizy"
        }
      ]
    },
    {
      "name": "2G",
      "changes": [
        {
          "lessons": {
            "first": 3,
            "last": 3
          },
          "info": "matematyka - Substitution: (Wypych Walerian) ➔ Słowiński Szczepan"
        }
      ]
    },
    {
      "name": "3A",
      "changes": [
        {
          "lessons": {
            "first": 6,
            "last": 6
          },
          "info": "(biologia) ➔ fizyka - Substitution: (Kalinowski Gracja) ➔ Kulik Olaf"
        }
      ]
    },
    {
      "name": "3B",
      "changes": []
    },
    {
      "name": "3C",
      "changes": [
        {
          "lessons": {
            "first": 3,
            "last": 3
          },
          "info": "religia - Substitution: (Rybarczyk Kalistrat) ➔ Jurek Arseniusz"
        },
        {
          "lessons": {
            "first": 5,
            "last": 5
          },
          "info": "historia - Substitution: (Szczygieł Robert) ➔ Pietruszka Dawid, za nieobecny oddział"
        }
      ]
    },
    {
      "name": "3E",
      "changes": []
    },
    {
      "name": "3F",
      "changes": [
        {
          "lessons": {
            "first": 6,
            "last": 6
          },
          "info": "historia - Substitution: (Szczygieł Robert) ➔ Wójcik Aleksandra"
        },
        {
          "lessons": {
            "first": 7,
            "last": 7
          },
          "info": "historia - Substitution: (Szczygieł Robert) ➔ Wysocki Harald, zł. grup"
        }
      ]
    },
    {
      "name": "3G",
      "changes": [
        {
          "lessons": {
            "first": 7,
            "last": 7
          },
          "info": "religia - Substitution: (Rybarczyk Kalistrat) ➔ Pietruszka Dawid, zł. grup"
        }
      ]
    },
    {
      "name": "4H",
      "changes": []
    }
  ]
}